test/jdk/javax/swing/JTable/JTableRightAlignmentTest.java

```diff
@@ -0,0 +1,217 @@
+                } else {
+                    System.out.println("Test Passed!");
+                }
+            } finally {
+                SwingUtilities.invokeAndWait(() -> {
+                    if (frame != null) {
+                        frame.dispose();
+                    }
+                });
+            }
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        try {
+            ImageIO.write(image, "png", new File(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+}
+
+class CustomTable {
+    public final static int COL_FIRSTNAME = 0;
+    public final static int COL_LASTNAME = 1;
+    public final static int COL_SALARY = 2;
+
+    static final Class[] classes = {
+            String.class,
+            String.class,
+            Float.class,
+    };
+
+    String[] cols = {
+            "First name",
+            "Last name",
+            "Salary",
+    };
+    List<CustomTable.Record> data = new ArrayList();
+    JTable table;
+
+    public CustomTable() {
+        data.add(new CustomTable.Record("First1", "Last1", 10000f));
+        data.add(new CustomTable.Record("First2", "Last2", 10000f));
+        data.add(new CustomTable.Record("First3", "Last3", 10000f));
+        table = new JTable(new CustomTable.Model());
+        table.getColumnModel().getColumn(COL_FIRSTNAME).setMaxWidth(90);
+        table.getColumnModel().getColumn(COL_LASTNAME).setMaxWidth(90);
+        table.getColumnModel().getColumn(COL_SALARY).setMaxWidth(90);
+    }
+
+    static class Record {
+        String firstname;
+        String lastname;
+        float salary;
+
+        public Record(String firstname, String lastname, float salary) {
+            this.firstname = firstname;
+            this.lastname = lastname;
+            this.salary = salary;
+        }
+    }
+
+    class Model extends AbstractTableModel {
+
+        public int getColumnCount() {
+            return cols.length;
+        }
+
+        public int getRowCount() {
+            return data.size();
+        }
+
+        public Object getValueAt(int rowIndex, int columnIndex) {
+            CustomTable.Record item = data.get(rowIndex);
+            switch (columnIndex) {
+                case COL_FIRSTNAME:
+                    return item.firstname;
+                case COL_LASTNAME:
+                    return item.lastname;
+                case COL_SALARY:
+                    return Float.valueOf(item.salary);
```

Comment:
```suggestion
                    return item.salary;
```
Explicit boxing is redundant.